scipy/stats/_distn_infrastructure.py

```diff
@@ -1611,18 +1611,19 @@ def nnlf(self, theta, x):
     def _nnlf(self, x, *args):
         return -np.sum(self._logpxf(x, *args), axis=0)
 
-    def _nnlf_and_penalty(self, x, args):
+    def _nlff_and_penalty(self, x, args, log_fitfun):
+        # negative log fit function
```

Comment:
The only real change is to use this to penalize `log_fitfun`, which can be `self._logpxf` or another function, instead of just `self._logpxf`. The rest is changing variable names so that they're not specific to MLE.